main.py

```diff
@@ -12,66 +12,66 @@
 )
 from makegif import gif_from_frames, gifsicle_optimize
 
-
-# working directory
-base_dir = os.path.abspath(os.path.dirname(__file__)) + "/"
-tmp_dir = "tmp/"
-out_dir = "out/"
-
-print_begin("Initializing directories")
-ensure_folder(tmp_dir)
-ensure_folder(out_dir)
-clear_folder(tmp_dir)
-
-os.chdir(base_dir + tmp_dir)
-print_check()
-
-
-fetch_frames(imgpath, cwd, framenaming)
-
-# sorted list of frames
-frames = sorted(
-    [file for file in next(os.walk(cwd))[2] if file.endswith("png")], key=str
-)
-
-
-# generate caption image
-caption = generate_caption_image(captiontext)
-caption = fit_caption_to_frame(Image.open(cwd + frames[0]), caption)
-
-
-# apply to each frame
-for i, frame in tqdm(enumerate(frames), desc="Applying caption", unit="frame"):
-    frame_img = Image.open(cwd + frame)
-
-    captioned = apply_caption(frame_img, caption)
-    captioned.save(cwd + frame)
-
-    frame_img.close()
-
-
-fname = generate_name(captiontext)
-
-if len(frames) <= 1:    
-    fname += ".png"
-    # todo: png optimization
-    print_begin(f"Moving result to {out_dir + fname}")
-    os.replace(cwd + frames[0], base_dir + out_dir + fname)
+if __name__ == '__main__':
```

Comment:
the proper way of doing this would be making a main function instead of putting all of the code inside the if

Reply:
Yeah I know but I'm not going to do too much work for boykissers like u :3

Reply:
😐